packages/graphql-yoga/src/plugins/resultProcessor/push.ts

```diff
@@ -20,9 +22,19 @@ export function processPushResult(
 
   let iterator: AsyncIterator<MaybeArray<ExecutionResult>>
 
+  let pingInterval: number
   const textEncoder = new fetchAPI.TextEncoder()
   const readableStream = new fetchAPI.ReadableStream({
-    start() {
+    start(controller) {
+      // ping client every 12 seconds to keep the connection alive
+      pingInterval = setInterval(() => {
+        if (!controller.desiredSize) {
+          clearInterval(pingInterval)
+          return
+        }
+        controller.enqueue(textEncoder.encode(':\n\n'))
+      }, timeoutInSeconds * 1000) as unknown as number
```

Comment:
Should this be based on the last emitted value instead? 12 seconds since the last message got sent to the client?

Reply:
Hmm makes sense to me. Let's do that.